src/charm.py

```diff
@@ -1,117 +1,282 @@
 
-        Learn more about config at https://juju.is/docs/sdk/config
+    def _on_get_certificate_action(self, event: ActionEvent) -> None:
+        """Triggered when users run the `get-certificate` Juju action.
 
         Args:
-            event: event triggering the handler.
+            event: Juju event
         """
-        # Fetch the new config value
-        log_level = self.model.config["log-level"].lower()
-
-        # Do some validation of the configuration option
-        if log_level in VALID_LOG_LEVELS:
-            # The config is good, so update the configuration of the workload
-            container = self.unit.get_container("httpbin")
-            # Verify that we can connect to the Pebble API in the workload container
-            if container.can_connect():
-                # Push an updated layer with the new config
-                container.add_layer("httpbin", self._pebble_layer, combine=True)
-                container.replan()
-
-                logger.debug("Log level for gunicorn changed to '%s'", log_level)
-                self.unit.status = ops.ActiveStatus()
-            else:
-                # We were unable to connect to the Pebble API, so we defer this event
-                event.defer()
-                self.unit.status = ops.WaitingStatus("waiting for Pebble API")
-        else:
-            # In this case, the config option is bad, so block the charm and notify the operator.
-            self.unit.status = ops.BlockedStatus("invalid log level: '{log_level}'")
-
-    @property
-    def _pebble_layer(self) -> pebble.LayerDict:
-        """Return a dictionary representing a Pebble layer."""
-        return {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {
-                        "GUNICORN_CMD_ARGS": f"--log-level {self.model.config['log-level']}"
-                    },
+        hostname = event.params["hostname"]
+        tls_certificates_relation = self._tls.get_tls_relation()
+        if not tls_certificates_relation:
+            event.fail("Certificates relation not created.")
+            return
+
+        tls_rel_data = tls_certificates_relation.data[self.app]
+        if tls_rel_data.get(f"certificate-{hostname}"):
+            event.set_results(
+                {
+                    f"certificate-{hostname}": tls_rel_data[f"certificate-{hostname}"],
+                    f"ca-{hostname}": tls_rel_data[f"ca-{hostname}"],
+                    f"chain-{hostname}": tls_rel_data[f"chain-{hostname}"],
                 }
-            },
-        }
+            )
+        else:
+            event.fail("Certificate not available")
```

Comment:
Just a nitpick:
If we do 
```
if not <condition>:
    event.fail()
    return
```
first, we get to reduce nesting for aesthetics :D

Reply:
good point, most of the tls logic was copied over from the nginx-ingress-integrator charm, so things like this can be overlooked!

I'll refactor this

Reply:
refactored!